ytsync/ytsync.go

```diff
@@ -493,10 +510,32 @@ func (s *Sync) processVideo(v video) (err error) {
 		log.Println(v.ID() + " took " + time.Since(start).String())
 	}(time.Now())
 
-	alreadyPublished, err := s.db.IsPublished(v.ID())
+	s.videosMapMux.Lock()
+	sv, ok := s.syncedVideos[v.ID()]
+	s.videosMapMux.Unlock()
+	alreadyPublished := ok && sv.Published
+
+	neverRetryFailures := []string{
+		"Error extracting sts from embedded url response",
+		"the video is too big to sync, skipping for now",
+	}
+	if ok && !sv.Published && util.SubstringInSlice(sv.FailureReason, neverRetryFailures) {
+		log.Println(v.ID() + " can't ever be published")
```

Comment:
how often does this happen, and do you know why? is it within our control?

Reply:
the first happens due to youtube restrictions (either region, age or deleted videos)
the second happens when we hit the defined size limit (it's in our control).

While we can't do anything about the first, we can raise the limits for the second, but as per several discussions we'll keep it at 2GB or less.

Reply:
out of 100k published videos we have about 1% that failed for those two (leading) reasons: https://scrn.storni.info/2018-08-09_12-24-24-339707393.png